x-pack/plugins/beats_management/public/lib/adapters/framework/kibana_framework_adapter.ts

```diff
@@ -55,71 +67,122 @@ export class KibanaFrameworkAdapter implements FrameworkAdapter {
     });
   };
 
-  public render = (component: React.ReactElement<any>) => {
-    this.rootComponent = component;
-  };
+  public waitUntilFrameworkReady(): Promise<void> {
+    return new Promise(resolve => {
+      this.onKibanaReady(async (Private: any, $injector: any) => {
+        const xpackInfo = Private(this.XPackInfoProvider);
+        let xpackInfoUnpacked: FrameworkInfo;
+        try {
+          xpackInfoUnpacked = {
+            basePath: this.getBasePath(),
+            k7Design: this.uiSettings.get('k7design'),
+            license: {
+              type: xpackInfo.getLicense().type,
+              expired: !xpackInfo.getLicense().isActive,
+              expiry_date_in_millis: xpackInfo.getLicense().expiryDateInMillis,
+            },
+            security: {
+              enabled: xpackInfo.get(`features.${this.PLUGIN_ID}.security.enabled`, false),
+              available: xpackInfo.get(`features.${this.PLUGIN_ID}.security.available`, false),
+            },
+            settings: xpackInfo.get(`features.${this.PLUGIN_ID}.settings`),
+          };
+        } catch (e) {
+          throw new Error(`Unexpected data structure from XPackInfoProvider, ${JSON.stringify(e)}`);
+        }
 
-  public hasValidLicense() {
-    if (!this.xpackInfo) {
-      return false;
-    }
-    return this.xpackInfo.get('features.beats_management.licenseValid', false);
-  }
+        const assertData = RuntimeFrameworkInfo.decode(xpackInfoUnpacked);
+        if (assertData.isLeft()) {
+          throw new Error(
+            `Error parsing xpack info in ${this.PLUGIN_ID},   ${PathReporter.report(assertData)[0]}`
+          );
+        }
+        this.xpackInfo = xpackInfoUnpacked;
 
-  public licenseExpired() {
-    if (!this.xpackInfo) {
-      return false;
-    }
-    return this.xpackInfo.get('features.beats_management.licenseExpired', false);
-  }
+        this.shieldUser = await $injector.get('ShieldUser').getCurrent().$promise;
 
-  public securityEnabled() {
-    if (!this.xpackInfo) {
-      return false;
-    }
+        const assertUser = RuntimeFrameworkUser.decode(this.shieldUser);
+        if (assertUser.isLeft()) {
+          throw new Error(
+            `Error parsing user info in ${this.PLUGIN_ID},   ${PathReporter.report(assertUser)[0]}`
+          );
+        }
 
-    return this.xpackInfo.get('features.beats_management.securityEnabled', false);
+        resolve();
+      });
+    });
   }
 
-  public getDefaultUserRoles() {
-    if (!this.xpackInfo) {
-      return [];
-    }
-
-    return this.xpackInfo.get('features.beats_management.defaultUserRoles');
+  public renderUIAtPath(
+    path: string,
+    component: React.ReactElement<any>,
+    toController: 'management' | 'self' = 'self'
+  ) {
+    const DOM_ELEMENT_NAME = this.PLUGIN_ID.replace('_', '-');
+    const adapter = this;
+    this.routes.when(
+      `${path}${[...Array(6)].map((e, n) => `/:arg${n}?`).join('')}`, // Hack because angular 1 does not support wildcards
```

Comment:
Maybe add `FIXME` before this comment, that angular 1 reference bring back so much memories. :D